src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -854,6 +864,35 @@ public virtual bool HasUnsupportedMetadata
             }
         }
 
+        internal bool? GetNonNullTypesFromSyntax()
+        {
+            bool? result = null;
+            string id = "CS" + ((int)ErrorCode.WRN_PragmaNonNullTypes).ToString("0000");
```

Comment:
Same comment about tying Symbol to error formatting.

Reply:
> Same comment about tying Symbol to error formatting.

This is how pragmas are handled. This code will go away once we switch to a dedicated directive.

---
In reply to: [224184123](https://github.com/dotnet/roslyn/pull/30387#discussion_r224184123) [](ancestors = 224184123)